### PR TITLE
[move-compiler] disable rewrite optimizations

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

This is an experiment to disable the optimizations I introduced during the CFG and HLIR  rewrites (#13497, https://github.com/MystenLabs/sui/pull/14641) until we have a better solution for handling on-chain divergence, in hopes of landing those rewrites sooner.
## Test Plan 

Revised tests match main, and all work.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: external-crates/move/crates/move-compiler/src/hlir/translate.rs [hunk 1]
```diff
@@ -1251,48 +1251,134 @@ fn value_block(
...
+// This is an slightly sub-optimal implementation of `value_list`, which we preserve in order to
+// accurately re-produce code that is already on-chain. When we support SHA-based source
+// verification, we should swap this version out with the optimized version below it. The main
+// difference is that this version does conversion and binding, then freezing; the other will
+// inline freezing when possible to avoid some bndings.
+
 fn value_list(
     context: &mut Context,
-    block: &mut Block,
+    result: &mut Block,
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/hlir/translate.rs:1262]: Would it be too much to gate this?

File: crates/sui-adapter-transactional-tests/tests/dynamic_fields/exhaustive.exp [hunk 2]
```diff
@@ -6,7 +6,7 @@ A: object(0,0)
 task 1 'publish'. lines 9-79:
 created: object(1,0)
 mutated: object(0,1)
-gas summary: computation_cost: 1000000, storage_cost: 12205600,  storage_rebate: 0, non_refundable_storage_fee: 0
+gas summary: computation_cost: 1000000, storage_cost: 12395600,  storage_rebate: 0, non_refundable_storage_fee: 0
```
Comment (tnowacki) [inline crates/sui-adapter-transactional-tests/tests/dynamic_fields/exhaustive.exp:9]: How does this compare to whats in main?

File: external-crates/move/crates/move-compiler/src/hlir/translate.rs (hunk 1 above)
Comment (cgswords) [reply external-crates/move/crates/move-compiler/src/hlir/translate.rs:1262]: I will try gating it, plus the other optimization.

File: crates/sui-adapter-transactional-tests/tests/dynamic_fields/exhaustive.exp (hunk 2 above)
Comment (cgswords) [reply crates/sui-adapter-transactional-tests/tests/dynamic_fields/exhaustive.exp:9]: This is the number on main.

**Review by tnowacki — APPROVED** [review]

**Pull request merged**